src/bundle/Resources/public/js/scripts/admin.contenttype.edit.js

```diff
@@ -213,21 +224,35 @@
             bodyData.FieldDefinitionCreate.position = draggedItemPosition;
         }
 
+        global.setTimeout(() => {
+            removeHighlight(addedField);
+        }, TIMEOUT_REMOVE_HIGHLIGHT);
+
         fetch(generateRequest('add', bodyData, languageCode))
             .then(ibexa.helpers.request.getTextFromResponse)
             .then((response) => {
-                removeLoadingField();
-                insertFieldDefinitionNode(response, { targetContainer, draggedItemPosition });
+                const shouldRemoveHighlight = loadingField.classList.contains(
+                    'ibexa-collapse--field-definition-highlight-marked-to-remove',
```

Comment:
This class was added as there  could be case that request is taking longer than three seconds - now after three seconds instead of removing highlight it's just marked to be removed and thanks to this class highlight is removed with the moment of inserting DOM with response from backend.